n5-viewer/README.md

```diff
@@ -1,2 +1,78 @@
+    * `cname` indicates the internal compression algorithm to be used. Must be one of the following: `lz4`, `lz4hc`, `blosclz`, `zstd`, `snappy`, or `zlib`.
+    * `nthreads` is a positive integer greater than or equal to 1, indicating the number of threads to use for compression.
+    * `shuffle` is an integer between -1 and 2 (inclusive) indicating the type of shuffling to perform, if any, prior to compression. 0 indicates no shuffling; 1 indicates byte-wise shuffling; 2 indicates bit-wise shuffling. -1 indicates the default procedure.
+    * `typesize` is a positive integer indicating the stride in bytes over which shuffling is performed. Required unless `shuffle` is 0, in which case the value is ignored.
+2. `bzip2`: If the type is `bzip2`, then the `compression` object must also contain the property `blockSize`. 
+    * See above definition of `blocksize`. Note that for `bzip2`, `blockSize` has a capital 'S'.
+3. `gzip`: If the type is gzip, then the compression object must also contain the properties `useZlib` and `level`.
+    * `useZlib` indicates whether or not Zlib was used. 
+    * `level` is equivalent to `clevel`, above.
+4. `lz4`: If the type is `lz4`, then the `compression` object must also contain the property `blockSize`.
+    * See above definition of `blocksize`. Note that for `lz4`, `blockSize` has a capital 'S'.
+5. `raw`: If the type is `raw`, this indicates no compression was or should be used.
+6. `xz`: If the type is `bzip2`, then the `compression` object must also contain the property `preset`.
+    * `preset` is an integer indicating the level of compression on a scale of -0 to -9. -0 is the lowest level of compression, and -9 is the highest. TODO: How to allow -0 in JSON Schema???
+
+Multiple compressors are not allowed. This property is required. 
+
+### blockSize
+A JSON array of positive integers indicating the length of each dimension of a chunk of the array. (Assumes the chunks are on a regular grid.) Must contain at least one item and cannot contain more than 4. This array must be the same length as `dimensions`, `pixelResolution`/`resolution`, `offset` (if present) and `downsamplingFactors` (if present), though this is not currently enforced by the schema. This property is required.
+
+### dimensions
+A JSON array of positive integers indicating the length of each dimension of the whole array. Must contain at least one item and cannot contain more than 4. This property is required.
+
+A couple of requirements are not currently enforced by the schema:
+1. The order of dimensions must be xyzt (or xyz, xyt, or xy). Dimension order is assumed by the software and is not expressed anywhere in the metadata, so a metadata document with incorrect dimension order will pass schema validation. 
+2. Channel is expressed in the hierarchy structure, not the metadata. Folders for each channel must be named e.g. c0/, c1/. 
```

Comment:
@bogovicj is it true that channel names are required to be named `c0`...`cN`? I know this is convention, at least